web/ui/react-app/src/hooks/useFetches.test.tsx

```diff
@@ -0,0 +1,55 @@
+import useFetches from './useFetches';
+import { renderHook } from '@testing-library/react-hooks';
+
+describe('useFetches', () => {
+  beforeEach(() => {
+    fetchMock.resetMocks();
+  });
+  it('should can handle multiple requests', async done => {
+    fetchMock.mockResponse(JSON.stringify({ satus: 'success', data: { id: 1 } }));
+    const { result, waitForNextUpdate } = renderHook(useFetches, { initialProps: ['/foo/bar', '/foo/bar', '/foo/bar'] });
+    await waitForNextUpdate();
+    expect(result.current.response).toHaveLength(3);
+    done();
+  });
+  it('should can handle success flow -> isLoading=true, response=[data, data], isLoading=false', async done => {
+    fetchMock.mockResponse(JSON.stringify({ satus: 'success', data: { id: 1 } }));
+    const { result, waitForNextUpdate } = renderHook(useFetches, { initialProps: ['/foo/bar'] });
+    expect(result.current.isLoading).toEqual(true);
+    await waitForNextUpdate();
+    expect(result.current.response).toHaveLength(1);
+    expect(result.current.isLoading).toEqual(false);
+    done();
+  });
+  it('should isLoading remains true on empty response', async done => {
+    fetchMock.mockResponse(jest.fn());
+    const { result, waitForNextUpdate } = renderHook(useFetches, { initialProps: ['/foo/bar'] });
+    expect(result.current.isLoading).toEqual(true);
+    await waitForNextUpdate();
+    setTimeout(() => {
+      expect(result.current.isLoading).toEqual(true);
+      done();
+    }, 1000);
```

Comment:
Is the `setTimeout()` here still needed after the `waitForNextUpdate()`? It seems to work without as well (but maybe just because the value is `true` to begin with)...

Reply:
Yes, It is working without the setTimeout. I just wanted to be clear that isLoading is checked after the response.